compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -1977,6 +1978,9 @@ void addGlobalVariable(DiagnosticPos pos, Set<Whitespace> ws, String identifier,
         }
         if (isListenerVar) {
             var.flagSet.add(Flag.LISTENER);
+            if (!isTypeNameProvided) {
+                var.isDeclaredWithVar = true;
+            }
```

Comment:
We can replace this with
```java
            var.isDeclaredWithVar = !isTypeNameProvided;
```
right?

Reply:
If the type name is provided we don't need to set "isDeclaredWithVar". IMO having the way suggested won't do anything but I think it will hide the actual logic.

Reply:
Was thinking if we could avoid evaluating/the if block. But it's that vs unnecessarily setting false for `var.isDeclaredWithVar`. Let's go with this.

Reply:
Sorry for dragging this conversation, but shouldn't this go into `generateBasicVarNode`. If `type name` is not provided I think we can consider it as declaredWithVar.